routers/routes/routes.go

```diff
@@ -102,6 +103,16 @@ func RouterHandler(level log.Level) func(ctx *macaron.Context) {
 	}
 }
 
+func createHeaderValueNew(maxAge time.Duration, sendPreloadDirective bool) string {
+	buf := bytes.NewBufferString("max-age=")
+	buf.WriteString(strconv.Itoa(int(maxAge.Seconds())))
+	buf.WriteString("; includeSubDomains")
```

Comment:
That means if somebody runs gitea on ```example.com``` HSTS policy would apply to ```*.example.com``` and their other subdomains will not load in a browser that uses HSTS  if they don't have SSL configured. So if they run another service on ```wiki.example.com``` that has no SSL it would stop working.

I think we should move this line to ```ifsendPreloadDirective``` check since it is required for preload but not required otherwise.

Reply:
done.

Reply:
If use preload then it has to meet the requirements for the other values:
https://hstspreload.org
https://wiki.mozilla.org/SecurityEngineering/HTTP_Strict_Transport_Security_(HSTS)_Preload_List

So Firefox bases their list from Google which requires maxAge >= 1 year and includeSubDomains, but then has separate rules to change it after. I think if a user sets ```SEND_PRELOAD_DIRECTIVE = true``` then we should force the other settings to match what Google requires and it will work in all browsers then.

Maybe like this:

```
if sendPreloadDirective {
    if len int(maxAge.Seconds()) < 31536000 {
      maxAge = time.Hour * 24 * 365
    }

    buf := bytes.NewBufferString("max-age=")
    buf.WriteString(strconv.Itoa(int(maxAge.Seconds())))
    buf.WriteString("; includeSubDomains")
    buf.WriteString("; preload")
} else {
    buf := bytes.NewBufferString("max-age=")
    buf.WriteString(strconv.Itoa(int(maxAge.Seconds())))
    if includeSubDomains {
        buf.WriteString("; includeSubDomains")
    }
}
```